packages/headless/src/features/result-preview/result-preview-request-builder.test.ts

```diff
@@ -32,7 +32,7 @@ describe('ResultPreviewRequestBuilder', () => {
       q: '',
       requestedOutputSize: 0,
       uniqueId: '1',
-      url: 'https://platform.cloud.coveo.com/rest/search/v2',
+      url: '',
```

Comment:
huh?

Reply:
The reason why the expected url is the empty string here is that we get the endpoint from the configuration initial state. I.e., no action is dispatched to set the search apiBaseUrl in these tests.

This kind of relates to the point you're making here https://github.com/coveo/ui-kit/pull/4339#discussion_r1754882593: it would be better to resolve the API base URLs when we build the requests.

Reply:
Gotcha, this was there in the test because the initial state already contained the url, got it!